eqcorrscan/utils/plotting.py

```diff
@@ -2184,6 +2184,122 @@ def subspace_fc_plot(detector, stachans, **kwargs):
     return fig
 
 
+@additional_docstring(plotting_kwargs=plotting_kwargs)
+def origintime_detections(detections, method='simple', **kwargs):
+
+    '''
+    Plot detections before and after applying match-filter and visual
+    comparison between them in one day.
+
+    :type detections: list
+    :param detections: list of eqcorrscan.core.match_filter.Detection
+    :type method: str
+    :param method: 'simple' or 'advance' defult to 'simple'
+    {plotting_kwargs}
+
+    :returns: :class:`matplotlib.figure.Figure`
+
+    .. warning::
+        For showing templates' origin, template name must be str of
+        the origin time of each template in match_filter or being changed to
+        origin time later.
+        Something like 2014-10-18T04:02:02
+        If template's name is something else, only detections will be plot.
+
+    .. rubric:: Example
+    >>> from eqcorrscan.core.match_filter import read_detections
+    >>> from eqcorrscan.utils.plotting import origintime_detections
+    >>> # Read text file detections
+    >>> detections = read_detections('path/to/file/filename')
+    >>> origintime_detections(detections, method='advance',
+    ...                       save=Ture, show=True,
+    ...                       savefile='./output.png')
+
+    .. rubric:: Example
+
+    >>> from eqcorrscan.core.match_filter import match_filter
+    >>> detections = match_filter(
+    ...     template_names=['2014-10-18T04:02:02'], template_list=[template],
+    ...     st=stream, threshold=10, threshold_type='MAD', trig_int=3,
+    ...     plot=False, output_cat=False, output_event=True)
+    >>> origintime_detections(detections, method='advance',
+    ...                       save=Ture, show=True,
+    ...                       savefile='./output.png')
+    '''
+    txt = '''
+        Can't show templates!
+        For showing templates' origin, template name must be str of
+        the origin time of each template in all detections.
+        Something like 2014-10-18T04:02:02
+        '''
+    import matplotlib.pyplot as plt
+    day = UTCDateTime(detections[0].detect_time.date)
+    times = [day.datetime, (day+(3600*24)).datetime]
+    plt.figure()
+    plt.plot(times, [0, 0])
+    if method == 'simple':
+        xdetects = [detect.detect_time.datetime for detect in detections]
```

Comment:
It looks like you are plotting detection-time, not origin-time, but your xlabel - origin time?

Reply:
Yes, i will change xlabel. thanks.